packages/sdk/src/evm/functions/contract/index.ts

```diff
@@ -0,0 +1,102 @@
+import { getCompositePluginABI } from "../../common/plugin";
+import { PREBUILT_CONTRACTS_MAP } from "../../contracts";
+import { SmartContract } from "../../contracts/smart-contract";
+import { ContractPublisher } from "../../core/classes/contract-publisher";
+import { getSignerAndProvider } from "../../core/classes/rpc-connection-handler";
+import { ValidContractInstance } from "../../core/types";
+import { AbiSchema } from "../../schema/contracts/custom";
+import {
+  cacheContract,
+  getCachedContract,
+  getCachedStorage,
+  inContractCache,
+} from "../core/cache";
+import { GetContractFromAbiParams, GetContractParams } from "./types";
+import { resolveContractType } from "./utils";
+
+export async function getContractFromAbi(
+  params: GetContractFromAbiParams,
+): Promise<ValidContractInstance> {
+  if (inContractCache(params.address)) {
+    return getCachedContract(params.address);
+  }
+
+  // TODO: Remove support for SDKOptions
+  const [signer, provider] = getSignerAndProvider(
+    params.network,
+    params.sdkOptions,
+  );
+
+  const parsedAbi =
+    typeof params.abi === "string" ? JSON.parse(params.abi) : params.abi;
+  const contract = new SmartContract(
+    signer || provider,
+    params.address,
+    await getCompositePluginABI(
+      params.address,
+      AbiSchema.parse(parsedAbi),
+      provider,
+      params.sdkOptions,
+      getCachedStorage(params.storage),
+    ),
+    getCachedStorage(params.storage),
+    params.sdkOptions,
+    (await provider.getNetwork()).chainId,
+  );
```

Comment:
you could optimize these to parallelize both the chainId resolution and the gompositepluginabi by calling them within a Promise.all before the `new SmartContract` and then passign them in sync